Assets/LeapMotion/Modules/InteractionEngine/Scripts/InteractionBehaviour.cs

```diff
@@ -357,6 +357,11 @@ public class InteractionBehaviour : MonoBehaviour, IInteractionBehaviour {
     /// If this event is fired on a given frame, it will occur after OnGraspEnd and before OnGraspStay.
     /// </remarks>
     public Action OnGraspBegin;
+    /// <summary>
+    /// Called when the object becomes grasped, if it was not already held by any interaction controllers on the
+    /// previous frame. Also passes itself as an event argument.
+    /// </summary>
+    public Action<InteractionBehaviour> OnObjectGraspBegin;
```

Comment:
Hmmm I'd be opposed to adding these extra callbacks unless there was really something you couldn't do without them.  There are already 2 different types of grasp callbacks, adding a third seems a little crowded.  

If you are subscribing to the event, you already know what object you are subscribing to, and could use that argument in your callback, right?

Reply:
I added these because there was really something I couldn't do without them: I had a manager class that sought to handle grasp events for the objects it managed. This is very important for designs involving multi-object grasping interactions, or interactions where there are multiple handles for a single object, where the logic to handle grasps should still be centralized. Without the grasped object passed as an argument, it's impossible to know which managed object was grasped, unless I did something weird like scanned through all objects to look for a just-changed "isGrasped" property.

Reply:
Couldn't you do something like this?

```
foreach(var obj in allTheObjs) {
  obj.OnGraspBegin += () =>  onObjGrasped(obj);
}

private void onObjGrasped(InteractionBehaviour obj) { //stuff }
```

Reply:
That.. is a good point.
It wouldn't work in a UnityEvent situation where you couldn't script up a lambda for it, and this does allocate-- but the allocation doesn't seem like a big deal.

I'll remove the callbacks.